src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -249,22 +225,26 @@ private void readRcaEnabledFromConf() {
    * has changed in the meantime (such as a new elected master). It also starts the RCA runtime if
    * it wasn't already running but the current state of the flag expects it to.
    */
-  private void startRcaNanny() {
+  public void startNanny() {
     netOpsExecutorService.scheduleAtFixedRate(
         () -> {
-          if (rcaScheduler != null && rcaScheduler.isRunning()) {
-            if (!rcaEnabled) {
-              // Need to shutdown the rca scheduler
-              stop();
-            } else {
+          netOpsExecutorService.schedule(this::readRcaEnabledFromConf, 1, TimeUnit.SECONDS);
```

Comment:
Let's not have calls to `schedule` inside `scheduleAtFixedRate`. We should have just the method references like it used to be. The expensive part was not that the another executor existed (they all use the same thread pool), but the call to `getElectedMasterHostAddress`.  So, let's just make the `nodeRolePoller` a no-op, is RCA is disabled.